core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -151,6 +155,7 @@ public FuzzyQueryBuilder(String fieldName, boolean value) {
      * @param value The value of the term
      */
     public FuzzyQueryBuilder(String fieldName, Object value) {
+        DEPRECATION_LOGGER.deprecated("{} query is deprecated. Instead use the [match] query with fuzziness parameter", NAME);
```

Comment:
I am on the fence. I think we should have a test for this, but I understand your objections. I am not sure we should rely on docs tests to verify this though. Docs tests are there to ensure that docs are correct, and we shouldn't be tempted to use them to verify that the code is correct. Ideally we should have a failing test besides docs if we remove the deprecation warning. Maybe add a unit test to fuzzy query builder tests that verifies that the deprecation log goes somewhere (in the ThreadContext). Is that feasible?